client/app/scripts/components/node-details.js

```diff
@@ -77,14 +77,19 @@ const NodeDetails = React.createClass({
     const details = this.props.details;
     const nodeExists = this.props.nodes && this.props.nodes.has(this.props.nodeId);
 
+    if (details) {
+      return this.renderDetails();
+    }
+
```

Comment:
Can we still `renderDetails` if `!nodeExists`? I'm specifically worried about that showing `undefined` in some spots...